app/models/miq_report/generator.rb

```diff
@@ -307,6 +307,13 @@ def generate_basic_results(options = {})
     # TODO: add once only_cols is fixed
     # targets = targets.select(only_cols)
     where_clause = MiqExpression.merge_where_clauses(self.where_clause, options[:where_clause])
+
+    # Remove custom_attributes as part of the `includes` if all of them exist
+    # in the select statement
+    if all_custom_attributes_are_virtual_sql_attributes?
+      remove_loading_relations_for_virtual_custom_attributes
+    end
```

Comment:
I want to understand this more:

```ruby
def _generate_table
  if ...
    4 generate_* methods
  else
    generate_basic_results # A
  end

  build_table # B
end
```

This change adds `remove_loading_relations_for_virtual_custom_attributes` A (optionally).
`build_table` always calls `remove_loading_relations_for_virtual_custom_attributes`.

1. Do we need to consider using that method for other `generate_*` methods?
2. Do we need to make it optional in `build_table`?

Reply:
> 1\. Do we need to consider using that method for other `generate_*` methods?

Possibly, though not the priority for this BZ.  Should function still without it at the moment, and I think the way we gather results for reports is vastly different in each of those methods, so I think waiting to do that makes sense for this PR.

> 2\. Do we need to make it optional in build_table?

No, this should be idempotent to remove the value from the `include` hash.  It is not saving the record with it removed anyway, so we are just removing it earlier than originally planed.